File: back_end/src/main/java/com/lukeshay/restapi/wall/Route.java
package com.lukeshay.restapi.wall;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
class Route {
  private String name;
  private String setter;
  private List<RouteTypes> types;
  private List<Grade> userGrade;
  private Grade averageGrade;
  private List<Integer> userRating;
  private Integer averageRating;
}

File: front_end/src/types/route.ts
export interface Route {
  id: string;
  wallId: string;
  gymId: string;
  name: string;
  setter: string;
  holdColor: string;
  types: Array<"BOULDER" | "TOP_ROPE" | "LEAD" | "AUTO_BELAY">;
  averageGrade: string;
  averageRating: number;
}
